fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -40,20 +40,64 @@ def all
         return @devices
       end
 
-      def clear_cache
-        @devices = nil
-      end
+      def connected_devices(requested_os_type)
+        Helper.log.info "Fetching available connected devices" if $verbose
 
-      # Reset all simulators of this type
-      def reset_all
-        all.each(&:reset)
+        device_types = []
+        if requested_os_type == "tvOS"
+          device_types = ["AppleTV"]
+        elsif requested_os_type == "iOS"
+          device_types = ["iPhone", "iPad"]
+        end
+
+        devices = [] # Return early if no supported devices are being searched for
+        if device_types.count == 0
+          return devices
+        end
+
+        usb_devices_output = ''
+        Open3.popen3("system_profiler SPUSBDataType") do |stdin, stdout, stderr, wait_thr|
+          usb_devices_output = stdout.read
+        end
+
+        device_uuids = []
+        usb_devices_output_lines = usb_devices_output.split(/\n/)
+        (0..usb_devices_output_lines.count).each do |line_index|
+          line = usb_devices_output_lines[line_index]
+          is_supported_device = !line.nil? && device_types.any? { |device_type| line.include?(device_type) }
+          next unless is_supported_device
+          usb_info_lines = 10
```

Comment:
This seems hacky. Instead you could use
```
system_profiler SPUSBDataType -xml
```
to get the output in XML so we can parse it. Please don't add `nokogiri` as a dependency, but instead use the `plist` gem :+1: